proto/cosmos/vesting/v1beta1/vesting.proto

```diff
@@ -83,3 +83,21 @@ message PermanentLockedAccount {
 
   BaseVestingAccount base_vesting_account = 1 [(gogoproto.embed) = true];
 }
+
+// TrueVestingAccount implements the VestingAccount interface. It provides
+// an account that can hold contributions subject to "lockup" (like a
+// PeriodicVestingAccount), or "true vesting" which is subject to clawback
+// of unvested tokens, or a combination (tokens vest, but are still locked).
+message TrueVestingAccount {
+  option (gogoproto.goproto_getters)  = false;
+  option (gogoproto.goproto_stringer) = false;
+
+  BaseVestingAccount base_vesting_account = 1 [(gogoproto.embed) = true];
+
+  // funder_address specifies the account which can perform clawback.
+  string funder_address = 2;
+  int64 start_time = 3 [(gogoproto.moretags) = "yaml:\"start_time\""];
+  repeated Period combined_periods = 4 [(gogoproto.moretags) = "yaml:\"combined_periods\"", (gogoproto.nullable) = false];
```

Comment:
They are explained here: https://github.com/Agoric/agoric-sdk/issues/4085#issuecomment-997600471

But I also don't find it very easy to follow.

Reply:
Can someone answer this:

Let's say someone's tokens are subject to 36 months vesting AND lockup from TGE, released every 6 months.
After month 6, is it:

- Option 1: 1/6th of "subject-to-lockup" tokens become vested, but only 1/6th of "subject-to-lockup" tokens are unlocked, which means that in fact they can only sell 1/36th of those tokens;

OR

- Option 2: the lockup runs concurrently to vesting; e.g. for all tokens regardless of vesting, it is the case that no more than x tokens shall be sold according to some lockup schedule. So after 6 months, 1/6th of tokens will be unlocked and unvested.

Or are both scenarios possible/configurable depending on the schedule of each? If so, how?

Reply:
I'll add comments here too. Combined periods gives the merge of the unlocking schedule and the vesting schedule, to determine when tokens are actually unencumbered. For instance, if I make a clawback grant of 1000 today (2022-01-17) that vests a 1/4th every 3months, but all of it is locked until Nov 1, then we'd have:

```
Unlocking schedule:
2022-11-01: 1000

Vesting schedule:
2022-04-17: 250
2022-07-17: 250
2022-10-17: 250
2023-01-17: 250

Combined schedule (when tokens are both vested and unlocked):
2022-11-01: 750
2023-01-17: 250
```

Per a note in `vesting_accounts.go`, I'm considering dropping the `combined_periods` field and just querying the two schedules directly. I'd expect the common use case is a vesting schedule with dozens of events, but with a lockup cliff, so this would be just as efficient.

Reply:
I think we can remove the combined schedule actually 👍

Reply:
Okay, I'll implement that.

BTW, this PR is fighting against the prevailing terminology in `x/auth/vesting` in that it uses "vesting" only to describe the status associated with clawback, and "lockup" to describe the encumbrance that prevents transfer - i.e. what is called "vesting" elsewhere. Let me know if I need to be even more emphatic about this in the comments.

Reply:
What would you have instead of the combined schedule then? I think it's important to keep it, because Option 1 of the comment above is unlikely to be the understanding of what most people mean by combined vesting and lockup, or the contractual terms.

Reply:
Oh, we'll have the same behavior, but we just won't precompute the combined schedule.  I.e. to get the unlocked coins at time `t`, instead of computing `ReadSchedule(combined_periods, t)`, we'd compute `min(ReadSchedule(lockup_periods, t), ReadSchedule(vesting_periods, t)`. (Not the exact API calls, of course, but that's the idea.)